cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -72,307 +78,383 @@ def __init__(
+                ]
             )
         )
 
         return cast(List[float], tlm_responses)
 
+    def _batch_get_trustworthiness_score_with_handled_errors(
+        self,
+        prompts: Sequence[str],
+        responses: Sequence[str],
+    ) -> List[Optional[float]]:
+        """Run batch of TLM get confidence score.
+
+        Args:
+            prompts (Sequence[str]): list of prompts to run get confidence score for
+            responses (Sequence[str]): list of responses to run get confidence score for
+
+        Returns:
+            List[float]: TLM confidence score for each prompt (in supplied order)
+        """
+        tlm_responses = self._event_loop.run_until_complete(
+            self._batch_async_with_handled_errors(
+                [
+                    self._get_trustworthiness_score_async(prompt, response)
+                    for prompt, response in zip(prompts, responses)
+                ]
+            )
+        )
+        return cast(List[Optional[float]], tlm_responses)
+
     async def _batch_async(
         self,
         tlm_coroutines: List[
             Union[Coroutine[None, None, TLMResponse], Coroutine[None, None, float]]
         ],
-        timeout: Optional[float],
     ) -> Union[List[TLMResponse], List[float]]:
         tlm_query_tasks = [asyncio.create_task(tlm_coro) for tlm_coro in tlm_coroutines]
 
-        return await asyncio.wait_for(asyncio.gather(*tlm_query_tasks), timeout=timeout)  # type: ignore[arg-type]
+        if self._verbose:
+            return await asyncio.wait_for(
+                tqdm_asyncio.gather(
+                    *tlm_query_tasks,
+                    total=len(tlm_query_tasks),
+                    desc="Querying TLM...",
+                    bar_format="{desc} {percentage:3.0f}%|{bar}|",
+                ),
+                timeout=self._timeout,
+            )
+        return await asyncio.wait_for(asyncio.gather(*tlm_query_tasks), timeout=self._timeout)  # type: ignore[arg-type]
+
+    async def _batch_async_with_handled_errors(
+        self,
+        tlm_coroutines: List[
+            Union[Coroutine[None, None, TLMResponse], Coroutine[None, None, float]]
+        ],
+    ) -> Union[List[Optional[TLMResponse]], List[Optional[float]]]:
+        tlm_query_tasks = [
+            asyncio.wait_for(asyncio.create_task(tlm_coro), timeout=self._timeout)
+            for tlm_coro in tlm_coroutines
+        ]
+
+        if self._verbose:
+            results = []
+            with tqdm(total=len(tlm_query_tasks)) as pbar:
+                for task in asyncio.as_completed(tlm_query_tasks):
+                    try:
+                        result = await asyncio.wait_for(task, timeout=self._timeout)
+                        results.append(result)
+                    except Exception:
+                        results.append(None)
+                    pbar.update(1)
+
+            return results  # type: ignore[return-value]
```

Comment:
An issue with this is that I'm sometimes getting this log:
```
Task exception was never retrieved
future: <Task finished name='Task-14' coro=<wait_for() done, defined at /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/asyncio/tasks.py:421> exception=TimeoutError()>
```

when running something like
```
tlm = studio.TLM( timeout=0.1)
tlm.try_prompt(["what is 1+2", "what is 3435+2"])
```

I think that is happening because of the for loop, but I am not sure how else to use `tqdm` and return exceptions at the same time (`tqdm_asyncio.gather` does not have a `return_exceptions` argument)